### PR TITLE
Misc fixes to issues found during docs building

### PR DESCRIPTION
**Pull request by Xarthisius**
## PR Summary

A few fixes for errors and warnings found by Jenkins

## PR Checklist

- [ ] pass `flake8 yt/`
- [ ] pass `isort . --check --diff`
- [ ] pass `black --check yt/`
- [ ] New features are documented, with docstrings and narrative docs
- [ ] Adds a test for any bugs fixed. Adds tests for new features.

### CODE REVIEW

File: doc/source/visualizing/callbacks.rst [hunk 1]
```diff
@@ -94,7 +94,7 @@ dataset from AMRVAC :
     import yt
 
     ds = yt.load("amrvac/bw_polar_2D0000.dat")
-    s = yt.plot2d(ds, 'density')
+    s = yt.SlicePlot(ds, axis=2, fields=['density'])
```
Comment (neutrinoceros) [inline doc/source/visualizing/callbacks.rst:97]: I think the explicit wart syntax is much better, especially in documentation, but I'm curious: why the change from plot_2d to SlicePlot ?

File: yt/visualization/color_maps.py [hunk 2]
```diff
@@ -508,7 +508,7 @@ def add_colormap(name, cdict):
 # Add colormaps in _colormap_data.py that weren't defined here
 _vs = np.linspace(0, 1, 256)
 for k, v in list(_cm.color_map_luts.items()):
-    if k not in yt_colormaps and k not in mcm.cmap_d:
+    if k not in yt_colormaps and k not in mcm._cmap_registry:
```
Comment (neutrinoceros) [inline yt/visualization/color_maps.py:511]: From what I've seen this is indeed the new place where this dict lives in mpl as of 3.3, but do you know if it's backward compatible?

File: doc/source/visualizing/callbacks.rst (hunk 1 above)
Comment (Xarthisius) [reply doc/source/visualizing/callbacks.rst:97]: It's simple, I couldn't find plot2d and I didn't even know that `plot_2d` existed :)

Comment (neutrinoceros) [reply doc/source/visualizing/callbacks.rst:97]: Oh nice, I'm only seeing the typo now, haha.
That's what code review is for I guess, always can use another pair of eyes !

File: yt/visualization/color_maps.py (hunk 2 above)
Comment (Xarthisius) [reply yt/visualization/color_maps.py]: fixed

File: doc/source/visualizing/callbacks.rst (hunk 1 above)
Comment (Xarthisius) [reply doc/source/visualizing/callbacks.rst]: reverted and fixed the typo

File: yt/visualization/color_maps.py (hunk 2 above)
Comment (neutrinoceros) [reply yt/visualization/color_maps.py]: Well done ! I wonder if we should refactor to just NOT use this dict since they are not keeping it exposed... but this is out of scope here.
